File: src/test/java/at/fhhagenberg/sqe/api/ElevatorControlSystemServiceTest.java
package at.fhhagenberg.sqe.api;

import at.fhhagenberg.sqe.di.DI;
import at.fhhagenberg.sqe.entity.Elevator;
import at.fhhagenberg.sqe.entity.ElevatorControlSystem;
import com.google.inject.Injector;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import java.rmi.RemoteException;
import static org.junit.jupiter.api.Assertions.*;

public class ElevatorControlSystemServiceTest {

    private ElevatorControlSystemService service;

    @BeforeEach
    public void setUp() {
        Injector injector = DI.getInjector();
        service = injector.getInstance(ElevatorControlSystemService.class);
    }

    @Test
    public void testGet() throws RemoteException {
        ElevatorControlSystem elevatorControlSystem = service.get();
        Elevator elevator = elevatorControlSystem.getElevator(0);

        assertNotNull(elevator);
        assertEquals(2, elevatorControlSystem.getNumberOfElevators());
        assertEquals(2, elevatorControlSystem.getNumberOfFloors());
        assertEquals(1, elevatorControlSystem.getClockTick());
        assertEquals(20, elevatorControlSystem.getFloorHeight());
        assertEquals(2, elevatorControlSystem.getNumberOfFloors());
        assertEquals(2, elevatorControlSystem.getFloors().size());
        assertEquals(0, elevatorControlSystem.getFloor(0).getFloorNumber());
        assertEquals(2, elevatorControlSystem.getNumberOfElevators());
        assertEquals(2, elevatorControlSystem.getElevators().size());
        assertEquals(0, elevatorControlSystem.getElevator(0).getElevatorNumber());
    }
}

File: src/test/kotlin/at/fhhagenberg/sqe/api/ElevatorControlSystemServiceTest.kt
package at.fhhagenberg.sqe.api

import at.fhhagenberg.sqe.di.TestDI
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test
import org.mockito.Mockito
import sqelevator.ConnectableIElevator
import sqelevator.IElevator
import java.rmi.RemoteException
import kotlin.test.*

class ElevatorControlSystemServiceTest {

    private lateinit var service: ElevatorControlSystemService
    private lateinit var realIElevator: IElevator

    @BeforeEach
    fun setUp() {
        val injector = TestDI.createMockInjector()
        service = injector.getInstance(ElevatorControlSystemService::class.java)
        realIElevator = injector.getInstance(ConnectableIElevator::class.java)
    }

    @Test
    @Throws(RemoteException::class)
    fun testGet() {
        val elevatorControlSystem = service.get()
        assertEquals(1, elevatorControlSystem.clockTick)
        assertEquals(20, elevatorControlSystem.floorHeight)
        assertEquals(2, elevatorControlSystem.numberOfFloors)
        assertEquals(2, elevatorControlSystem.floors.size)
        assertEquals(0, elevatorControlSystem.getFloor(0)?.floorNumber)
        assertNull(elevatorControlSystem.getFloor(-1))
        assertEquals(2, elevatorControlSystem.numberOfElevators)
        assertEquals(2, elevatorControlSystem.elevators.size)
        assertEquals(0, elevatorControlSystem.getElevator(0)?.elevatorNumber)
        assertNull(elevatorControlSystem.getElevator(-1))
    }

    @Test
    @Throws(RemoteException::class)
    fun testGetNegativeElevators() {
        Mockito.`when`(realIElevator.elevatorNum).thenReturn(-1)
        val elevatorControlSystem = service.get()
        assertEquals(0, elevatorControlSystem.numberOfElevators)
        assertEquals(0, elevatorControlSystem.elevators.size)
        assertNull(elevatorControlSystem.getElevator(0))
    }

    @Test
    @Throws(RemoteException::class)
    fun testGetNegativeFloors() {
        Mockito.`when`(realIElevator.floorNum).thenReturn(-1)
        val elevatorControlSystem = service.get()
        assertEquals(0, elevatorControlSystem.numberOfFloors)
        assertEquals(0, elevatorControlSystem.floors.size)
        assertNull(elevatorControlSystem.getFloor(0))
    }
}